src/main/scala/nl.knaw.dans.easy.deposit/servlets/DepositServlet.scala

```diff
@@ -159,49 +157,28 @@ class DepositServlet(app: EasyDepositApiApp)
       ).getOrRecoverResponse(respond)
   }
 
-  private def uploadFileItem(uuid: UUID, path: Path, uploadItem: FileItem): Try[(Boolean, FileItem)] = {
+  private def uploadFileItem(uuid: UUID, path: Path, uploadItem: FileItem): Try[FileItem] = {
+    managed(uploadItem.getInputStream).apply[Try[_]] {
+      case is if isZip(uploadItem) => app.unzipDepositFile(is, uploadItem.charset, user.id, uuid, path)
+      case is => app.writeDepositFile(is, user.id, uuid, path.resolve(uploadItem.name))
+    }.map(_ => uploadItem)
+  }
+
+  private def isZip(uploadItem: FileItem) = {
     val extensionIsZip = uploadItem.name.matches(".*.g?z(ip)?")
     lazy val contentTypeIsZip = uploadItem.contentType.exists(_.matches(
       "(application|multipart)/(x-)?g?zip(-compress(ed)?)?( .*)?"
     ))
-    logger.debug(s"is it a ZIP? ${ uploadItem.name } : $extensionIsZip; ${ uploadItem.contentType } : $contentTypeIsZip ")
-    if (extensionIsZip || contentTypeIsZip)
-      managed(uploadItem.getInputStream)
-        .apply(uploadZip(uuid, path, uploadItem, _))
-        .map((_, uploadItem))
-    else {
-      val fullPath = path.resolve(uploadItem.name)
-      managed(uploadItem.getInputStream)
-        .apply(app.writeDepositFile(_, user.id, uuid, fullPath))
-        .map((_, uploadItem))
-    }
+    logger.debug(s"ZIP check: ${ uploadItem.name } : $extensionIsZip; ${ uploadItem.contentType } : $contentTypeIsZip ")
+    extensionIsZip || contentTypeIsZip
   }
 
-  private def uploadZip(uuid: UUID, path: Path, uploadItem: FileItem, uploadInputStream: InputStream): Try[Boolean] = {
-    val zipInputStream = uploadItem.charset
-      .map(charSet => new ZipInputStream(uploadInputStream, Charset.forName(charSet)))
-      .getOrElse(new ZipInputStream(uploadInputStream))
-
-    def handleZipEntry(zipEntry: ZipEntry) = {
-      if (zipEntry.isDirectory) Success(false)
-      else {
-        logger.info(s"extracting ${ zipEntry.getName } size=${ zipEntry.getSize } crc=${ zipEntry.getCrc } method=${ zipEntry.getMethod }")
-        val fullPath = path.resolve(zipEntry.getName)
-        app.writeDepositFile(zipInputStream, user.id, uuid, fullPath)
-      }
+  private def getContentTypeIfMultipart = {
+    val multiPart = "multipart/"
+    request.getHeader("Content-Type").blankOption match {
```

Comment:
would it look better if you would do this match using a `map` and [`getOrRecover`](https://github.com/DANS-KNAW/dans-scala-lib/blob/c7edb07a0331ecc3ffaacebc9a058408154278b5/src/main/scala/nl/knaw/dans/lib/error/package.scala#L160)?